scan/lib/scan/slack_poster.rb

```diff
@@ -55,7 +56,11 @@ def run(results)
         attachment_properties: {
           fields: fields
         }
-      })
+      }
+      options = FastlaneCore::Configuration.create(Fastlane::Actions::SlackAction.available_options, arguments)
+      result = Fastlane::Actions::SlackAction.run(options)
+      return arguments if Helper.test? # Used to help validate the arguments passed to SlackAction
```

Comment:
As you mentioned this does not sounds good since this changes API interface only for testing, so I would suggest leveraging rspec-mock APIs to inspect expected arguments is given to `FastlaneCore::Configuratio` to compose correct options. Please find attached.

https://gist.github.com/ainame/8ae7a0ec329091747330cfdbb63e6a17

Reply:
Aahhhhhhh I was missing `.and_call_original` 😱 🔥 😱 🔥 😱  
Thank you!! 😁😁 I spent a lot of time trying to make the "expect" to work with no luck, cuz the "any_args" (or, in my case I was using "anything") was sending nil to the original method instead 😅 

Sweet, I'll update 😊

Reply:
Done!

Reply:
Just wondering: I was using `anything` before instead of `any_args`. They sound like the same thing, are they any different? 🤔

Reply:
> expect(double).to receive(:msg).with(1, any_args) # any args acts like an arg splat and can go anywhere
> expect(double).to receive(:msg).with(1, anything(), "b") #2nd argument can be anything at all

According to the doc, `any_args` can fulfil multiple args whilst anything doesn't? I think `anything` would be better if it works. I have just never used and known `anything` before 😅

Reply:
Ahh gotcha! Yeah makes sense haha 😊 thanks! 
And apparently `any_args` can be used in both cases yeah 😁 there must be some drawback of using `any_args` when `anything` would suffice, but I couldn't find anything online to answer that question 🤔